image/v1/tests/ImageStreamImportMode.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "Image"
+crdName: images.config.openshift.io
+tests:
+  onCreate:
+   - name: Should be able to create a minimal Image config
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Image
+        spec: {}
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Image
+        spec: {}
+   - name: Should be able to configure imageStreamImportMode to Legacy
```

Comment:
Add a test for the empty string as well

Reply:
i'm wondering how we can do this as the idea is that - when there is no import mode specified, it will be inferred based on the payload type and will be set by the image registry operator. When/where would these tests run?

Reply:
Well this is just an API integration test, so, you'd test that it is accepted as empty. Whatever mutation mechanism you choose to implement would be separate from these tests anyway. That test will likely want to be in origin.

When you say set by the operator, are you planning to set the spec? Typically that's not done and you re-calculate. If it's status, you can set it.

Reply:
It is the status. If there is no value specified in the spec or it is "", the status would be updated to `Legacy` or `PreserveOriginal` based on the payload type that CVO advertises (single vs multi payload)

Reply:
> Whatever mutation mechanism you choose to implement would be separate from these tests anyway

So iiuc, this would just test the API standalone? not like an e2e?

Reply:
added the test for empty string.